app/models/location.rb

```diff
@@ -1,6 +1,9 @@
 class Location < ApplicationRecord
   belongs_to :organisation
+
   has_many :ips
+  validates_associated :ips
```

Comment:
Without this, it's possible to save locations with invalid IPs, though it seems to be inconsistent when.

With it, there's some kind of bug that seems to lead to an infinite validation loop sometimes - a bad enough bug that I don't want to merge this without fixing it.

Reply:
Had a chat with Tom and we're starting to track this down.

Looks like it's a combination of the custom validation on Ip (`validate :address_must_be_valid_ip`) and the validation declaration on its parent (`validates_associated :ips`). Digging into it.

We also discovered that the problem only occurs when at least two IPs - one invalid and one valid - are saved side-by-side. 

If `validates_associated` is removed, an invalid IP can be saved following the same pattern - saving a location with one valid IP and one invalid IP.